examples/five_mem_node/main.rs

```diff
@@ -0,0 +1,321 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#[macro_use]
+extern crate log;
+extern crate env_logger;
+extern crate protobuf;
+extern crate raft;
+extern crate regex;
+
+use std::collections::{HashMap, VecDeque};
+use std::sync::mpsc::{self, Receiver, Sender, SyncSender, TryRecvError};
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant};
+use std::{str, thread};
+
+use protobuf::Message as PbMessage;
+use raft::storage::MemStorage;
+use raft::{prelude::*, StateRole};
+use regex::Regex;
+
+fn main() {
+    env_logger::init();
+
+    // Create 5 mailboxes to send/receive messages.
+    let (mut tx_vec, mut rx_vec) = (Vec::new(), Vec::new());
+    for _ in 0..5 {
+        let (tx, rx) = mpsc::channel();
+        tx_vec.push(tx);
+        rx_vec.push(rx);
+    }
+
+    // A pending proposals queue.
+    let proposals = Arc::new(Mutex::new(VecDeque::<Proposal>::new()));
+
+    let mut handles = Vec::new();
+    for (i, rx) in rx_vec.into_iter().enumerate() {
+        let proposals = Arc::clone(&proposals);
+        let mailboxes = (1..6u64).zip(tx_vec.iter().cloned()).collect();
+        let mut node = match i {
+            0 => Node::create_raft_leader(1, rx, mailboxes),
+            _ => Node::create_raft_follower((i + 1) as u64, rx, mailboxes),
+        };
+
+        let mut t = Instant::now();
+        let handle = thread::spawn(move || loop {
```

Comment:
It's good in example code to write a bit more comments than you might normally, as the user is just getting started with the project. For example "Here we spawn the node on a new thread and keep a handle so we can join on them later."